src/Symfony/Component/DependencyInjection/Tests/Compiler/IntegrationTest.php

```diff
@@ -411,6 +413,34 @@ public function testLocatorConfiguredViaAttribute()
         self::assertSame($container->get(BarTagClass::class), $s->locator->get(BarTagClass::class));
         self::assertSame($container->get(FooTagClass::class), $s->locator->get('with_key'));
         self::assertFalse($s->locator->has('nullable'));
+        self::assertSame('foo', $s->locator->get('subscribed'));
+    }
+
+    public function testIteratorConfiguredViaAttribute()
+    {
+        $container = new ContainerBuilder();
+        $container->setParameter('some.parameter', 'foo');
+        $container->register(BarTagClass::class)
+            ->setPublic(true)
+        ;
+        $container->register(FooTagClass::class)
+            ->setPublic(true)
+        ;
+        $container->register(AutowireIteratorConsumer::class)
+            ->setAutowired(true)
+            ->setPublic(true)
+        ;
+
+        $container->compile();
+
+        /** @var AutowireIteratorConsumer $s */
+        $s = $container->get(AutowireIteratorConsumer::class);
+
+        self::assertIsArray($s->iterator);
```

Comment:
Is this correct? I expected this to be a `RewindableGenerator`.

Reply:
And let AutowirePass unwrap when only an array is accepted, right?

Reply:
It was my understanding that a tagged iterator argument is _always_ injected as `RewindableGenerator` (regardless of the type-hint) so it's lazy. In the docs we show using `iterable` for the type-hint. In other tests it's clearly a `RewindableGenerator`, I'm not sure why in this case it isn't.